mu/modes/circuitpython.py

```diff
@@ -45,10 +45,19 @@ class CircuitPythonMode(MicroPythonMode):
         (0x1209, 0xBAB1, None, "Electronic Cats Meow Meow"),
         (0x1209, 0xBAB2, None, "Electronic Cats CatWAN USBStick"),
         (0x1209, 0xBAB3, None, "Electronic Cats Bast Pro Mini M0"),
+        (0x1209, 0xBAB6, None, "Electronic Cats Escornabot Makech"),
         (0x1B4F, 0x8D22, None, "SparkFun SAMD21 Mini Breakout"),
         (0x1B4F, 0x8D23, None, "SparkFun SAMD21 Dev Breakout"),
         (0x1209, 0x2017, None, "Mini SAM M4"),
         (0x1209, 0x7102, None, "Mini SAM M0"),
+        (0x04D8, 0xEC72, None, "XinaBox CC03"),
+        (0x04D8, 0xEC75, None, "XinaBox CS11"),
+        (0x04D8, 0xED5E, None, "XinaBox CW03"),
```

Comment:
@KalbeAbbas are these PIDs sublicense from Microchip? Or are they share with Microchip or other products?

Reply:
Btw, these PID/VID have only been added to CircuitPython, is this what you expected? Do these boards also support standard MicroPython? If so, which mode would you prefer to support?

Reply:
How are the CC03 and CS11 modules connected to the computer? Is there a USB module that needs to be plugged? If so, is that module just routing USB pins into a USB connector, or do they have additional USB device?
I couldn't find any info on the CW03, do you have any links?

https://xinabox.cc/products/cc03
https://xinabox.cc/products/cs11

Reply:
A programmer is necessary, either Xinabox IP02 or Xinabox IP03. @KalbeAbbas has written a guide here: https://www.hackster.io/PragmaticPhil/loading-circuitpython-onto-an-xinabox-programmable-core-544749

Thus it might be that it's actually the IP02/IP03 which are recognized, but I don't believe these programmers are used for anything else than Xinabox devices.

I can't find info on the CW03 either, I guess it's no longer in production.

MicroPython seems to support the SAMD21 processor: https://github.com/micropython/micropython/tree/master/ports/samd

Reply:
So it looks like IP03 is just a breakout for the SWD/JTAG pins on the microcontroller modules.

IPO2 actually has an FTD232RL (by default 0x0403, 0x6001), so while they could be getting these VID/PID preprogrammed from factory by Microchip, I'm not sure how the FTDI module would change these values depending on the microcontroller module plugged. Also I think ths might be just a USB-to-UART chip.

I guess it is quite possible the IP03 module also has USB data pass-through to the microcontroller modules.

Maybe @pragmaticphil (Hi Phil! 👋 ) could also point us to the CW03 device info?

Reply:
**Re: CS11 and CC03**
Yes, the CS11 and CC03 is only supported via CircuitPython, not MicroPython. They have their own VID/PID and I believe has to be included like this:
```
add /Applications/mu-editor.app/Contents/Resources/app/mu/modes/circuitpython.py line 49:
    (0x04D8, 0xEC72),  # XinaBox CC03
    (0x04D8, 0xEC75),  # XinaBox CS11
```
the CW03 is not ready yet, so let us forget about that one for now. And correct, the IP03 just connected standard Micro-USB connector to the USB via a small connector to the CS11/CC03. It does not have any USB device on it. 

**Re: CW01 and CW02**
These two are using UART to USB chip, being the standard FTDI chip, with its own VID/PID combination. The FTDI chip sits on a separate module, either IP01 or IP02, they both use the same FTDI chip. The CW01 is based on the ESP8266 and the CW02 is based on the ESP32, and they both support the standard MicroPython implementation.

I believe the way these are being supported is like this
```
add to modes/esp.py line 47:
     (0x0403, 0x6001)   # XinaBox CW01, CW02 (FTDI) 
```

Hope this clarifies things a bit.

Reply:
Hi @gotfredsen, thanks for the clarification. However, how devices are added to Mu has changed slightly, for example see https://github.com/mu-editor/mu/blob/master/mu/modes/circuitpython.py#L39

Or the proposed changes in this PR: https://github.com/mu-editor/mu/pull/1094/commits/b1764ec2ef4ef754a59ebc92ffd43825bd65d6db

When this PR is merged "XinaBox CC03" and "XinaBox CS11" will show up in the user-interface, so the user can see the name of the connected devices, however for CW01/CW02 we will not be able to detect which device it is, so they will just show up as "CircuitPython device" or similar, as long as they don't have their own VID/PID or a unique manufacturer-string.

Reply:
@dybber, I have accepted the propose change in PR #b1764ec

And with regards to CW01, CW02 and the use FTDI, is that also ok?

Reply:
That change is also part of it, I forgot to link to it: https://github.com/mu-editor/mu/pull/1094/commits/75b65c577c3bc5459040c27b71b84b3319062a30#diff-e78b8abb0346fee662c3583eb227d9a6R54

Reply:
The remaining questions is what @carlosperate wrote to begin with: 
> Are these PIDs sublicense from Microchip? Or are they shared with Microchip or other products? 

(For CC03, CS11, CW03)

Reply:
@dybber, I have accepted the propose change in PR #75b65c5 as well.

The PIDs are sublicensed from Microchip and are not shared with any other devices, not ours nor others.

Reply:
Okay, I guess we can mark this as resolved then. Thanks!

Reply:
Thank you @gotfredsen that definitely helps a lot!

I believe the `0x04D8` VID belongs to Microchip, and they do sublicense PIDs to third parties, but unfortunately there doesn't seem to be a public list online to be able to check this.
Are these PIDs sublicensed for these products?  Just want to make sure there might not be clashes with other USB devices with the same values.

Reply:
My PR view wasn't refreshed when I sent my last message (I'm used to GH keeping the tabs synced, but that didn't happen this time).
My question has been answer already, sorry for the duplication!

Reply:
> Thank you @gotfredsen that definitely helps a lot!
> 
> I believe the `0x04D8` VID belongs to Microchip, and they do sublicense PIDs to third parties, but unfortunately there doesn't seem to be a public list online to be able to check this.
> Are these PIDs sublicensed for these products? Just want to make sure there might not be clashes with other USB devices with the same values.

@carlosperate, the PIDs we got from Microchip are sub-licensed to us and are not used with any other device, neither from us, nor from anybody else. They are unique to our CS11, CC03 and CW03.